uint32 width
uint32 height
uint32 roi_offset_x
uint32 roi_offset_y
uint32 roi_width
uint32 roi_height
bool flip
